plugins/GSdx/boost_spsc_queue.hpp

```diff
@@ -169,6 +195,17 @@ class ringbuffer_base
         return write_index_.is_lock_free() && read_index_.is_lock_free();
     }
 
+    bool size() const
+    {
+        const size_t write_index =  write_index_.load(memory_order_relaxed);
+        const size_t read_index = read_index_.load(memory_order_relaxed);
+        if (read_index > write_index) {
+            return (write_index + max_size) - read_index;
+        } else {
+            return write_index - read_index;
```

Comment:
Well better spend time to redo the MTVU atomic management. My initial port to std::atomic was completely crazy ! And architecture of the code isn't good for atomic.

Code ought to be
```
T1
wp = load write position (memory relaxed)
do write 1 on wp
do write 2 on wp + x
commit write position (memory acquire); // act as a memory store barrier
```
So far we reload the atomic every write with random memory constrains (and same for the read part)